frontend/src/pages/user/admin/allReports/index.tsx

```diff
@@ -0,0 +1,76 @@
+import Form from 'react-bootstrap/Form';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom';
+import FormItem from '@components/FormItem';
+
+interface Props {
+  year: number;
+  month: number;
+}
+
+const AdminReport = () => {
+  const navigate = useNavigate();
+  const baseYear = 2020;
+  const currentYear = new Date().getFullYear();
+  const years = Array.from(
+    { length: currentYear - baseYear + 1 },
+    (_, index) => currentYear - index,
+  );
+  const months = Array.from({ length: 12 }, (_, index) => index + 1);
+
+  const { register, handleSubmit } = useForm<Props>({
+    defaultValues: {
+      year: years[0],
+      month: months[0],
+    },
+  });
+
+  const onSubmit: SubmitHandler<Props> = (data) => {
+    navigate(`/admin/reports/${data.year}/${data.month}`);
+  };
+
+  return (
+    <div style={{ padding: '5% 8% 10% 8% ' }}>
+      <h3>Select report period</h3>
+      <hr className='hr' />
+
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <FormItem label='Year'>
+          <Form.Select
+            {...register('year', {
+              setValueAs: (value) => parseInt(value),
+            })}
```

Comment:
is the ``parseInt`` necessary

Reply:
yes because the value put in html will automatically change to string, 
when you lack of the parseInt() and submit with initial value, it will looks like this
`Object { year: "2021", month: "6" }`, in order to let the type be correct, the parseInt() is needed